### ISSUE TITLE
Unable to create private playlist

### ISSUE THREAD

**Issue by thomir** [spotify/web-api#903]
Issue found on June 1st 2018.

When creating a new playlist (with the `POST` endpoint), even when sending `public: false` in the JSON body, the playlist is created with `public: true`. 

#### Endpoint(s):
* `POST https://api.spotify.com/v1/users/{user_id}/playlists`

#### Scope(s):
* This is reproducible with all scopes selected.

#### Steps to reproduce - creating a new playlist:
1. Open the API Console to the [post-playlist endpoint](https://developer.spotify.com/console/post-playlists/?user_id=thomir&body=%7B%22name%22%3A%22New%20Playlist%22%2C%22description%22%3A%22New%20playlist%20description%22%2C%22public%22%3Afalse%7D)
2. Call the `/v1/users/{user_id}/playlists` endpoint. I created an oauth token with all available scopes checked, and changed the request body to contain `"public": false` (see screenshot:)
![image](https://user-images.githubusercontent.com/372783/40815228-17acf8dc-6599-11e8-81cc-2412da848f65.png)

#### Expected behaviour - creating a new playlist:
I expect the playlist to be created with `public: false`. Instead, the returned playlist object has `public: true` - see screenshot:
![image](https://user-images.githubusercontent.com/372783/40815269-6c310056-6599-11e8-9f4a-3319832200f1.png)

**Comment by hughrawlinson** [comment]
Hi! We pushed a fix for this issue. Please let me know if you're still able to reproduce it, and I'll reopen the ticket. Thanks for reporting!

**Comment by thomir** [comment]
Hi @hughrawlinson ,

Thanks for the fix, and sorry it took me so long to get around to testing it!

It looks like this is mostly fixed, however I notice that there's still something funny going on with the Spotify desktop client. To reproduce:

* Create a new playlist using [the web console](https://developer.spotify.com/console/post-playlists/) with `"public": false`
* Note that the returned playlist is in fact private (yay!)
* Open Spotify desktop client, right click on the newly created playlist.
* Note that the menu item says "make secret" - even though the playlist is already "secret".
 * If you click 'make secret', nothing changes in the playlist according to the API. If you then click 'make public', the playlist does indeed toggle to being public.

If I had to guess, it seems like the desktop client sets the initial state for all playlists it hasn't seen before to be public by default, and ignores the `public` attribute from the API.

I'm not sure if this should be a separate bug - it seems like it's not actually an API bug (retrieving the playlist details are consistent with the information we specified when creating the playlist), but rather a client bug. I don't see anywhere obvious to report this however.

Thanks again for the fix!
